From 8d46803c09edc2b6d4e35c778a3d2f90e5baad0b Mon Sep 17 00:00:00 2001
From: Maxim Cournoyer <maxim.cournoyer@gmail.com>
Date: Sat, 25 Mar 2023 22:55:16 -0400
Subject: [PATCH] Fallback to elogind when systemd is unavailable at build time
 https://bugs.webkit.org/show_bug.cgi?id=254475

Reviewed by NOBODY (OOPS!).

The build system supports elogind, but it only considers the
'libsystemd' library name for the pkg-config lookup and not
'libelogind'.  This change makes the build system fallback to search
for libelogind when libsystemd was not found.

* Source/cmake/FindJournald.cmake [!PC_SYSTEMD_FOUND]: Search for libelogind.
---
 Source/cmake/FindJournald.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/cmake/FindJournald.cmake b/Source/cmake/FindJournald.cmake
index 18dd6b50908c..e0dc9ce1d397 100644
--- a/Source/cmake/FindJournald.cmake
+++ b/Source/cmake/FindJournald.cmake
@@ -55,6 +55,10 @@ find_package(PkgConfig QUIET)
 
 # libelogind provides compatible pc and header files
 pkg_check_modules(PC_SYSTEMD QUIET libsystemd)
+if (NOT PC_SYSTEMD_FOUND)
+    pkg_check_modules(PC_SYSTEMD QUIET libelogind)
+endif ()
+
 set(Journald_COMPILE_OPTIONS ${PC_SYSTEMD_CFLAGS_OTHER})
 set(Journald_VERSION ${PC_SYSTEMD_VERSION})
 

base-commit: 43ea8744bc6065aad7ae5988e32d31d253905e5f
-- 
2.39.2

